src/REPLMode/argument_parsers.jl

```diff
@@ -71,22 +71,20 @@ end
 # packages can be identified through: uuid, name, or name+uuid
 # additionally valid for add/develop are: local path, url
 function parse_package_identifier(word::AbstractString; add_or_develop=false)::PackageSpec
-    if add_or_develop && casesensitive_isdir(expanduser(word))
+    if add_or_develop && Types.isdir_windows_workaround(expanduser(word))
```

Comment:
@KristofferC It seems that `casesensitive_isdir` is not valid for `.` and `..`. Things were working before because we were just accepting it in the "guess it is a url" part. Given that we have to check the casing of directories in the API layer anyway, I just moved it out REPL layer.